actnow/accounts/views.py

```diff
@@ -1,33 +1,29 @@
-from django.utils.translation import gettext_lazy as _
-from rest_framework import status
-from rest_framework.generics import CreateAPIView
-from rest_framework.response import Response
+from django.contrib.auth import get_user_model
+from rest_framework import viewsets
 
-from actnow.accounts.serializers import UserRegistrationSerializer
+from actnow.accounts.serializers import UserSerializer
 
-from .permissions import AllowAppicationOwnerOnly
-from .utils import email_address_exists, username_exists
+from .permissions import AllowAppicationOwnerOnly, IsAuthenticated, IsOwnerOrReadOnly
 
 
-class UserRegistrationView(CreateAPIView):
-    serializer_class = UserRegistrationSerializer
-    permission_classes = (AllowAppicationOwnerOnly,)
+class UsersView(viewsets.ModelViewSet):
+    queryset = get_user_model().objects.all()
+    serializer_class = UserSerializer
 
-    def create(self, request, *args, **kwargs):
-        serializer = self.get_serializer(data=request.data)
-        serializer.is_valid(raise_exception=True)
-        if email_address_exists(serializer.data["email"]):
-            response = _("A user is already registered with this e-mail address.")
-            response_status = status.HTTP_400_BAD_REQUEST
-            return Response({"error": response}, status=response_status)
+    def get_permissions(self):
+        if self.action == "create":
+            permission_classes = (AllowAppicationOwnerOnly,)
+        else:
+            permission_classes = (IsOwnerOrReadOnly, IsAuthenticated)
+        return [permission() for permission in permission_classes]
```

Comment:
Not sure I follow here

Reply:
We only want `Applications` to create a new account right? That's why we are assigning `AllowAppicationOwnerOnly` permission to `create` action. For the other actions e.g `update, delete`, we ensure that it's the resource owner trying to modify the resource.

Reply:
My bad, wasn't clear: I follow the logic but I don't get why we need to initialize `permission_classes` first using tuples and then convert them to list, etc. Why not just go with lists to begin with?

Reply:
I see. Not much difference I'll update it.